connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/DockerJDBCIntegrationV2Suite.scala

```diff
@@ -19,10 +19,17 @@ package org.apache.spark.sql.jdbc.v2
 
 import java.sql.Connection
 
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.jdbc.DockerJDBCIntegrationSuite
 
 abstract class DockerJDBCIntegrationV2Suite extends DockerJDBCIntegrationSuite {
 
+  override def sparkConf: SparkConf = super.sparkConf
+    // DS V2 relies on ANSI mode to translate expressions, we should always
+    // run JDBC v2 tests with ANSI on.
+    .set(SQLConf.ANSI_ENABLED, true)
```

Comment:
A better approach is to skip the entire test suite in the non ANSI CI job, but I can't figure out a way to dynamically disable a test suite with the current scalatest framework we use.